pydantic/networks.py

```diff
@@ -298,11 +298,13 @@ def __get_validators__(cls) -> 'CallableGenerator':
         if email_validator is None:
             raise ImportError('email-validator is not installed, run `pip install pydantic[email]`')
 
-        yield str_validator
         yield cls.validate
 
     @classmethod
-    def validate(cls, value: str) -> 'NameEmail':
+    def validate(cls, value: Any) -> 'NameEmail':
+        if type(value) == cls:
+            return value
+        value = str_validator(value)
         return cls(*validate_email(value))
```

Comment:
@samuelcolvin 
> Looks good, could you please also fix NameEmail and add a test for that too.
I found a bug that the validator can't parse NameEmail. It means `NameEmail` filed doesn't accept an instance of `NameEmail`